example/bcachefs.nix

```diff
@@ -188,61 +125,40 @@
         mountOptions = [
           "verbose"
         ];
-        mountpoint = "/";
         subvolumes = {
-          "subvolumes/rootfs" = { };
-          "subvolumes/home" = { };
-          "subvolumes/home/user" = { };
-          "subvolumes/nix" = { };
-          "subvolumes/test" = { };
+          # Subvolume name is different from mountpoint.
+          "subvolumes/root" = {
+            mountpoint = "/";
+          };
+          # Subvolume name is the same as the mountpoint.
+          "subvolumes/home" = {
+            mountpoint = "/home";
+          };
+          # Nested subvolume doesn't need a mountpoint as its parent is mounted.
+          "subvolumes/home/user" = {
+          };
+          # Parent is not mounted so the mountpoint must be set.
+          "subvolumes/nix" = {
+            mountpoint = "/nix";
+          };
+          # This subvolume will be created but not mounted.
+          "subvolumes/test" = {
+          };
         };
       };
 
-      # # Example showing mounted subvolumes in a multi-disk configuration (not yet working).
-      # mounted_subvolumes_in_multi = {
-      #   type = "bcachefs_filesystem";
-      #   passwordFile = "/tmp/secret.key";
-      #   extraFormatArgs = [
-      #     "--compression=lz4"
-      #     "--background_compression=lz4"
-      #   ];
-      #   mountOptions = [
-      #     "verbose"
-      #   ];
-      #   subvolumes = {
-      #     # Subvolume name is different from mountpoint
-      #     "foo" = {
-      #       mountpoint = "/bar";
-      #     };
-      #     # Subvolume name is the same as the mountpoint
-      #     "home" = {
-      #       mountpoint = "/home";
-      #     };
-      #     # Sub(sub)volume doesn't need a mountpoint as its parent is mounted
-      #     "home/user" = {
-      #     };
-      #     # Parent is not mounted so the mountpoint must be set
-      #     "nix" = {
-      #       mountpoint = "/nix";
-      #     };
-      #     # This subvolume will be created but not mounted
-      #     "test" = {
-      #     };
-      #   };
-      # };
-
       # Example showing another bcachefs filesystem.
       no_reliance_on_external_subvolume = {
         type = "bcachefs_filesystem";
         mountpoint = "/sometestdir";
```

Comment:
I just realised that this particular example doesn't make sense, since this mount point actually does rely on another one in a different subvolume. Tomorrow (or later today, I should say), I'll remove it or maybe move it to a separate test file altogether.

Reply:
Resolved by eae05708d8217dab44ef4e38deeaa73a5008d5f4.